omod/src/main/java/org/openmrs/module/spa/SpaController.java

```diff
@@ -95,38 +71,28 @@ public ResponseEntity<Resource> getStaticFile(HttpServletRequest request) {
             }
 
             return ResponseEntity.ok()
-                    .cacheControl(CacheControl.maxAge(180, TimeUnit.DAYS))
-                    .headers(headers ->
-                            headers.setExpires(Instant.now().plus(180, ChronoUnit.DAYS))
-                    )
+                    .cacheControl(CacheControl.noCache().mustRevalidate())
                     .contentType(mediaType)
                     .body(resource);
         } else {
-            return ResponseEntity.notFound().build();
+            return ResponseEntity.notFound().cacheControl(CacheControl.noCache().mustRevalidate()).build();
         }
     }
 
     /**
-     * This controller method should handle everything else, and we assume that the correct response is the index.html
-     * file. Like {@link #getStaticFileWithoutCacheHeaders(String)}, this attempts to force the client to always reload
-     * this resource.
+     * This controller method should handle every other URL. We assume that the correct response is the index.html, i.e.,
+     * the single page.
      */
     @RequestMapping({ "/", "/**/{filename:.?(?!.*\\.[^.]*$).*$}", "*.html", "*.htm"})
     public ResponseEntity<Resource> getSinglePage() {
         Resource resource = resourceLoader.getResource("/index.html");
         if (resource.exists()) {
             return ResponseEntity.ok()
                     .cacheControl(CacheControl.noCache().mustRevalidate())
-                    .header("Expires", "0")
                     .contentType(MediaType.TEXT_HTML)
                     .body(resource);
         } else {
-            return ResponseEntity.notFound().build();
+            return ResponseEntity.notFound().cacheControl(CacheControl.noCache().mustRevalidate()).build();
         }
     }
-
-    @RequestMapping("")
-    public ResponseEntity<Resource> index(HttpServletRequest request) {
-        return ResponseEntity.status(HttpStatus.MOVED_PERMANENTLY).location(URI.create(request.getContextPath() + "/spa/")).build();
-    }
```

Comment:
Not sure why this got removed, but I'm also not sure why it was here in the first place, so defer to your knowledge and understanding of all of this.

Reply:
Good catch.

Reply:
I had locally been playing around with trying to replace this with the `getSinglePage()` end point, but it wasn't fully working. Removing it in this PR is a mistake.